controllers/network/network_operator.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package network
+
+import (
+	"github.com/avinetworks/sdk/go/models"
+	"github.com/pkg/errors"
+	"github.com/vmware-samples/load-balancer-operator-for-kubernetes/pkg/aviclient"
+)
+
+type UsableNetworks []UsableNetwork
+
+type UsableNetwork struct {
+	NetworkName string `json:"networkName"`
+	CIDR        string `json:"cidr"`
+}
+
+type UsableNetworkOperator struct{}
+
+func (c *UsableNetworkOperator) AddUsableNetwork(client aviclient.Client, cloudName, networkName string) (bool, error) {
+	network, err := client.NetworkGetByName(networkName)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to get Data Network %s from AVI Controller\n", networkName)
+	}
+	cloud, err := client.CloudGetByName(cloudName)
+	if err != nil {
+		// Cannot find the configured cloud, requeue the request but
+		// leave enough time for operators to resolve this issue
+		return false, errors.Wrapf(err, "Failed to find cloud %s, requeue the request\n", cloudName)
+	}
+	if cloud.IPAMProviderRef == nil {
+		// Cannot find any configured IPAM Provider, requeue the request but
+		// leave enough time for operators to resolve this issue
+		return false, errors.Wrap(err, "No IPAM Provider is registered for the cloud, requeue the request")
+	}
+	ipamProviderUUID := aviclient.GetUUIDFromRef(*(cloud.IPAMProviderRef))
+	ipam, err := client.IPAMDNSProviderProfileGet(ipamProviderUUID)
+	if err != nil {
+		return false, errors.Wrap(err, "Failed to find IPAM profile")
+	}
+
+	// Ensure network is added to the cloud's IPAM Profile as one of its
+	// usable Networks
+	foundUsableNetwork := false
+	for _, usableNetwork := range ipam.InternalProfile.UsableNetworks {
+		if *usableNetwork.NwRef == *(network.URL) {
+			foundUsableNetwork = true
+			break
+		}
+	}
+	if foundUsableNetwork {
```

Comment:
This is wrong, if already found the usable network in the IPAM, why do you still want to add it?

The condition should be the opposite way. You should look this part: https://github.com/vmware-samples/load-balancer-operator-for-kubernetes/blob/main/controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go#L256 more carefully.

Reply:
lol, great catch! this indicates my e2e was trivially passing. forgot to add `Should(BeTrue())` after `Eventually`